src/core/ddsi/src/ddsi_typebuilder.c

```diff
@@ -769,10 +770,17 @@ static dds_return_t typebuilder_add_union (struct typebuilder_data *tbd, struct
     }
     if (is_default)
     {
+      def_member = &tb_aggrtype->detail._union.cases[c];
       tb_aggrtype->detail._union.cases[c].is_external = is_ext;
       tb_aggrtype->detail._union.cases[c].is_default = true;
       tb_aggrtype->detail._union.cases[c].is_last_label = true;
-      tb_aggrtype->detail._union.cases[c].disc_value = 0;
+      /* In case of integer disciriminator type, set value for default label to 0. For enum
```

Comment:
You're right, I didn't realize that it is not used. In [8188262](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1433/commits/818826239abcb467bd99956d3a03e79a7da82bd9) I've changed the IDL C generator so that it always outputs `0` for the discriminant value of the default case in the opcodes. The loop to get the first available value in the type builder is also removed.